lesson18.md

```diff
@@ -0,0 +1,385 @@
+
+---
+
+### Quiz
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+
+console.log(friends[1]); // ???
+console.log(friends[3]); // ???
+```
+
+```js
+console.log(friends[1]); // "Bob"
+console.log(friends[3]); // undefined
+```
+<!-- .element: class="fragment" -->
+
+
+
+### Exercise 1
+
+* Step 1: Create an array with your 3 top friends
+* Step 2: Say "hello" on console to each friend, e.g.:
+
+```plaintext
+hello Alice
+hello Bob
+hello Carol
+```
+
+---
+
+### Solution
+
+```js
+let friends = ["Sevtap", "Carlo", "Abdullah"];
+console.log("hello " + friends[0]);
+console.log("hello " + friends[1]);
+console.log("hello " + friends[2]);
+```
+
+---
+
+### Modifying arrays
+
+We can change any value using brackets:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+
+friends[1] = "David";
+// friends array is now ["Alice", "David", "Carol"]
+
+console.log(friends[1]); // "David"
+```
+
+---
+
+### Common operations: array length
+
+We can get the length of an array with the `.length` property:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+console.log(friends.length); // 3
+```
+
+---
+
+### Iterating through an array
+
+How can we make the code below more generic, for any array size?
+
+```js
+let cars = ["Saab", "Volvo", "BMW"];
+
+console.log(cars[0]);
+console.log(cars[1]);
+console.log(cars[2]);
+```
+
+---
+
+### Loops!
+
+```js
+let cars = ["Saab", "Volvo", "BMW"];
+
+for (let i = 0; i < cars.length; i++) {
+    console.log(cars[i]);
+}
+```
+
+---
+
+### Exercise 2
+
+* Step 1: Create an array with your 3 top friends (you can use the same from exercise 1)
+* Step 2: Say "hello" on console to each friend, using a `for` loop!
+
+```
+hello Alice
+hello Bob
+hello Carol
+```
+
+---
+
+### Appending new values
+
+We can append new values to an array using `.push()`:
+
+```js
+let friends = ["Alice", "Bob", "Carol"];
+console.log(friends.length); // 3
+
+friends.push("David");
+
+console.log(friends.length); // 4
+console.log(friends[3]); // David
+```
+
+---
+
+### Exercise 3
+
+Create a function that takes an array and outputs the following:
+
+* Output the last element of the array
+* Output the sum of all the numbers in the array
+* Output the largest number of the array
+* Output the average (mean) of all numbers (sum of all numbers divided by amount of numbers)
+
+Create an array containing a few numbers, ex `[3, 2, 9, 5, 6]` and test your code:
+
+```js
+function exercise3(array) {
+    // your code here
+}
+exercise3([3, 2, 9, 5, 6]);
+```
+
+<!-- .slide: style="font-size:70%" -->
+
+---
+
+### Bonus Exercise
+
+* Create a new array containing only numbers of the original array that are smaller than 5
+ * ex: given input `[1, 3, 7, 2, 9, 5]`, return `[1, 3, 2]`
+* Write a function that takes a number and an array of numbers as input parameter. Return the index of the number if it is in the array, otherwise return `-1`
+ * ex: given inputs `5` and `[2, 1, 5, 6, 3]`, return `2`
+
+---
+
+### Solution
```

Comment:
Have we had any feedback on solutions in the slides? My concern would be that it makes it too easy for them to just look ahead to get the solutions.

Reply:
yes, I was thinking the same. I'll remove them tomorrow